source/swap/docs/Swap.md

```diff
@@ -0,0 +1,197 @@
+
+
+
+### <span id="Swap-authorizeSigner-address-"></span> `authorizeSigner(address authorizedSigner)` (external)
+
+Authorize a delegated signer
+
+
+Emits an AuthorizeSigner event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSigner`|address Address to authorize
+
+
+
+### <span id="Swap-revokeSender-address-"></span> `revokeSender(address authorizedSender)` (external)
+
+Revoke an authorized sender
+
+
+Emits a RevokeSender event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSender`|address Address to revoke
+
+
+
+### <span id="Swap-revokeSigner-address-"></span> `revokeSigner(address authorizedSigner)` (external)
+
+Revoke an authorized signer
+
+
+Emits a RevokeSigner event
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizedSigner`|address Address to revoke
+
+
+
+### <span id="Swap-isSenderAuthorized-address-address-"></span> `isSenderAuthorized(address authorizer, address delegate) → bool` (internal)
+
+Determine whether a sender delegate is authorized
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizer`|address Address doing the authorization
+|`delegate`|address Address being authorized
+
+
+|  Param    | Description   |
+|   ---------| :------------ |
+| `bool`|  True if a delegate is authorized to send
+
+### <span id="Swap-isSignerAuthorized-address-address-"></span> `isSignerAuthorized(address authorizer, address delegate) → bool` (internal)
+
+Determine whether a signer delegate is authorized
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`authorizer`|address Address doing the authorization
+|`delegate`|address Address being authorized
+
+
+|  Param    | Description   |
+|   ---------| :------------ |
+| `bool`|  True if a delegate is authorized to sign
+
+### <span id="Swap-isValid-struct-Types-Order-bytes32-"></span> `isValid(struct Types.Order order, bytes32 domainSeparator) → bool` (internal)
+
+Validate signature using an EIP-712 typed data hash
+
+
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`order`|Types.Order Order to validate
+|`domainSeparator`|bytes32 Domain identifier used in signatures (EIP-712)
+
+
+|  Param    | Description   |
+|   ---------| :------------ |
+| `bool`|  True if order has a valid signature
+
+### <span id="Swap-transferToken-address-address-uint256-address-bytes4-"></span> `transferToken(address from, address to, uint256 param, address token, bytes4 kind)` (internal)
+
+Perform an ERC-20 or ERC-721 token transfer
+
+
+Transfer type specified by the bytes4 kind param
+ERC721: uses transferFrom for transfer
+ERC20: Takes into account non-standard ERC-20 tokens.
+
+| Param             | Description       |
+| ----------------- | :------------    |
+|`from`|address Wallet address to transfer from
```

Comment:
Is there a way to split the type out into its own column?

Reply:
next iteration, working on modifying solidity-docgen.